ompi/mca/coll/libpnbc_osc/pnbc_osc.c

```diff
@@ -435,6 +500,79 @@ static inline int NBC_Start_round(NBC_Handle *handle) {
 
     memcpy (&type, ptr, sizeof (type));
     switch(type) {
+      //TODO
+      case PUT:
+        NBC_DEBUG(5,"  PUT (offset %li) ", offset);
+        NBC_GET_BYTES(ptr,putargs);
+        NBC_DEBUG(5,"*buf: %p, count: %i, type: %p, dest: %i, tag: %i)\n", putargs.buf,
+                  putargs.count, putargs.datatype, putargs.dest, handle->tag);
+        /* get an additional request */
+        handle->req_count++;
+        /* get buffer */
+        if(sendargs.tmpbuf) {
+          buf1=(char*)handle->tmpbuf+(long)sendargs.buf;
+        } else {
+          buf1=(void *)sendargs.buf;
+        }
+#ifdef NBC_TIMING
+        Iput_time -= MPI_Wtime();
+#endif
+        tmp = (MPI_Request *) realloc ((void *) handle->req_array, handle->req_count * sizeof (MPI_Request));
+        if (NULL == tmp) {
+          return OMPI_ERR_OUT_OF_RESOURCE;
+        }
+        
+        handle->req_array = tmp;
+        res = handle->win->w_osc_module->osc_put(buf1, putargs.count, putargs.datatype,
+                                         putargs.dest, 0, putargs.count,
+                                         putargs.datatype, handle->win);
+
+        if (OMPI_SUCCESS != res) {
+          NBC_Error ("Error in MPI_Iput(%lu, %i, %p, %i, %i, %lu) (%i)", (unsigned long)buf1, putargs.count, putargs.datatype, putargs.dest, handle->tag, (unsigned long)handle->comm, res);
+          return res;
+        }
+#ifdef NBC_TIMING
+        Iget_time += MPI_Wtime();
+#endif
+        break;
+     case GET:
+        NBC_DEBUG(5,"  GET (offset %li) ", offset);
+        NBC_GET_BYTES(ptr,getargs);
+        NBC_DEBUG(5,"*buf: %p, count: %i, type: %p, dest: %i, tag: %i)\n", getargs.buf,
+                  getargs.count, getargs.datatype, getargs.source, handle->tag);
+        /* get an additional request */
+        handle->req_count++;
+        /* get buffer */
+        if(sendargs.tmpbuf) {
+          buf1=(char*)handle->tmpbuf+(long)sendargs.buf;
+        } else {
+          buf1=(void *)sendargs.buf;
+        }
+#ifdef NBC_TIMING
+        Iget_time -= MPI_Wtime();
+#endif
+        //TODO: I am not too sure we need to realloc for PUT/GET - Not used
+        tmp = (MPI_Request *) realloc ((void *) handle->req_array, handle->req_count * sizeof (MPI_Request));
+        if (NULL == tmp) {
+          return OMPI_ERR_OUT_OF_RESOURCE;
+        }
+        
+        handle->req_array = tmp;
+
+        res = handle->win->w_osc_module->osc_get(buf1, getargs.count, getargs.datatype,
+                                         getargs.source, 0, getargs.count,
```

Comment:
```suggestion
                                         getargs.target, 0, getargs.count,
```
See comment for putargs.dest above.

Reply:
Done